app/src/interfaces/input-rich-text-html/useImage.ts

```diff
@@ -115,12 +115,12 @@ export default function useImage(
 		imageDrawerOpen.value = false;
 	}
 
-	function onImageSelect(image: Record<string, any>) {
-		const assetUrl = getPublicURL() + 'assets/' + image.id;
+	function onImageSelect(image: File) {
+		const assetUrl = getPublicURL() + 'assets/' + image.filename_disk;
```

Comment:
I don't think we can guarantee that the `filename_disk` is always going to be the generated ID + extension, even if that's what Directus does on new uploads. I've seen uses in the past where people update it on the adapter and in the database directly. 

The assets endpoint expects `/assets/<primary key>`, and accepts any string of characters after the primary key. I think the safer bet here is to keep it as `'assets/' + image.id` and then use the `extension` function form `mime-types` to generate the correct extension for the `directus_files.type` field.

Reply:
That makes sense that we cannot guarantee file extension from the `filename_disk`. I will update the change to include the `mime-type` from the `extension` function.

Reply:
Hey @rijkvanzanten I wasn't able to use the `mime-type` package as it caused a circular dependency. 

I used a combination of the `mime/lite` to pull from the the file `type` field, but that seems like it can be null, and have a fall back to check the `filename_download` as it should have an extension to be able to render properly when downloaded.